src/sentry/sentry_metrics/indexer/ratelimiters.py

```diff
@@ -0,0 +1,136 @@
+from typing import Mapping, Optional, Sequence, Tuple
+
+from django.conf import settings
+
+from sentry import options
+from sentry.ratelimits.sliding_windows import (
+    GrantedQuota,
+    Quota,
+    RedisSlidingWindowRateLimiter,
+    RequestedQuota,
+    Timestamp,
+)
+from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.indexer.base import KeyCollection, KeyResult
+from sentry.utils import metrics
+
+writes_limiter = RedisSlidingWindowRateLimiter(
+    **settings.SENTRY_METRICS_INDEXER_WRITES_LIMITER_OPTIONS
+)
+
+
+OrgId = int
+
+
+def _build_quota_key(use_case_id: UseCaseKey, org_id: Optional[OrgId] = None) -> str:
+    use_case_str = {
+        UseCaseKey.PERFORMANCE: "performance",
+        UseCaseKey.RELEASE_HEALTH: "releasehealth",
+    }[use_case_id]
+
+    if org_id is not None:
+        return f"metrics-indexer-{use_case_str}-org-{org_id}"
+    else:
+        return f"metrics-indexer-{use_case_str}-global"
+
+
+@metrics.wraps("sentry_metrics.indexer.construct_quotas")  # type: ignore
+def _construct_quotas() -> Mapping[UseCaseKey, Sequence[Quota]]:
+    """
+    Construct write limit's quotas based on current sentry options.
+
+    This value can potentially cached globally as long as it is invalidated
+    when sentry.options are.
+    """
+    return {
+        UseCaseKey.PERFORMANCE: [
+            Quota(prefix_override=_build_quota_key(UseCaseKey.PERFORMANCE, None), **args)
+            for args in options.get("sentry-metrics.writes-limiter.limits.performance.global")
+        ]
+        + [
+            Quota(prefix_override=None, **args)
+            for args in options.get("sentry-metrics.writes-limiter.limits.performance.per-org")
+        ],
+        UseCaseKey.RELEASE_HEALTH: [
+            Quota(prefix_override=_build_quota_key(UseCaseKey.RELEASE_HEALTH, None), **args)
+            for args in options.get("sentry-metrics.writes-limiter.limits.releasehealth.global")
+        ]
+        + [
+            Quota(prefix_override=None, **args)
+            for args in options.get("sentry-metrics.writes-limiter.limits.releasehealth.per-org")
+        ],
+    }
+
+
+@metrics.wraps("sentry_metrics.indexer.construct_quota_requests")  # type: ignore
+def _construct_quota_requests(
+    use_case_id: UseCaseKey, keys: KeyCollection
+) -> Tuple[Sequence[OrgId], Sequence[RequestedQuota]]:
```

Comment:
Why returning a tuple of sequences? It seems that returning a `Mapping[OrgId, RequestedQuota]` would make it harder to break.